File: Vesti/Vesti/AsyncImageView.m
//
//  AsyncImageView.m
//  Vesti
//
//  Created by Djuro Alfirevic on 4/26/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

#import "AsyncImageView.h"

@implementation AsyncImageView

#pragma mark - Properties

- (void)setImageURL:(NSString *)imageURL {
    _imageURL = imageURL;
    
    dispatch_queue_t downloadQueue = dispatch_queue_create("ImageDownloader", NULL);
    dispatch_async(downloadQueue, ^{
        NSData *data = [NSData dataWithContentsOfURL:[NSURL URLWithString:self.imageURL]];
        UIImage *image = [[UIImage alloc] initWithData:data];
        
        dispatch_async(dispatch_get_main_queue(), ^{
            self.image = image;
        });
    });
}

@end

File: VestiS/VestiS/AsyncImageView.swift
//
//  AsyncImageView.swift
//  VestiS
//
//  Created by Djuro Alfirevic on 8/3/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

import UIKit

class AsyncImageView: UIImageView {

    // MARK: - Properties
    var imageURL: String? {
        didSet {
            if let imageURL = imageURL {
                dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), {
                    if let url = NSURL(string: imageURL) {
                        if let data = NSData(contentsOfURL: url) {
                            if let image = UIImage(data: data) {
                                dispatch_async(dispatch_get_main_queue(), {
                                    self.image = image
                                })
                            }
                        }
                    }
                })
            }
        }
    }
}